swift/apple/FirezoneKit/Sources/FirezoneKit/Models/Favorites.swift

```diff
@@ -1,45 +1,51 @@
 import Foundation
 
 public class Favorites: ObservableObject {
-  private static let key = "favoriteResourceIDs"
+  private let key = "favoriteResourceIDs"
   @Published private(set) var ids: Set<String>
 
-  public init() {
-    ids = Favorites.load()
+  public init () {
+    self.ids = Set()
+    Task { await self.ids = load() }
   }
 
   func contains(_ id: String) -> Bool {
     return ids.contains(id)
   }
 
-  func reset() {
-    objectWillChange.send()
-    ids = Set()
-    save()
+  func load() async -> Set<String> {
+    let ids = await withCheckedContinuation { continuation in
+      continuation.resume(returning: UserDefaults.standard.stringArray(forKey: key))
+    }
+
+    if let ids {
+      return Set(ids)
+    }
+
+    return Set()
   }
 
-  func add(_ id: String) {
-    objectWillChange.send()
-    ids.insert(id)
-    save()
+  func reset() async {
+    self.ids = Set()
+    await save()
   }
 
-  func remove(_ id: String) {
-    objectWillChange.send()
-    ids.remove(id)
-    save()
+  func add(_ id: String) async {
+    self.ids.insert(id)
+    await save()
```

Comment:
If `save` fails, your in-memory state will drift from disk state. If you make this fn async, I would suggest to try to save to disk first and then update the in-memory model.

Reply:
I haven't changed anything here - just a little bit of refactoring for a consistent use of `self`.

The `save()` call doesn't throw, so if it did error we wouldn't know about 🤷